### PR TITLE
[MRG] Overlap verification code.

### LINKED ISSUE #755: Add `allow_overlap` in NiftiMapsMasker, NiftiSpheresMasker
In NiftiMapsMasker, some people want to prevent any overlap. Such option can be useful. On the other hand, some people may want to allow overlapping spheres in NiftiSpheresMasker.

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
Check for overlap in MapsMasker and SpheresMasker.

Fix #755.

- [x] Code
- [x] Doc
- [x] Tests

### CODE REVIEW

File: nilearn/input_data/nifti_spheres_masker.py [hunk 1]
```diff
@@ -17,7 +17,8 @@
 from .base_masker import filter_and_extract, BaseMasker
 
 
-def _iter_signals_from_spheres(seeds, niimg, radius, mask_img=None):
+def _iter_signals_from_spheres(seeds, niimg, radius, mask_img=None,
+                               allow_overlap=False):
```
Comment (bthirion) [inline nilearn/input_data/nifti_spheres_masker.py]: Do you thinks that this should be the default. Maybe this deserves a word in the documentation.

**Comment by bthirion** [comment]
A test for this new feature would be great...

File: nilearn/input_data/nifti_spheres_masker.py (hunk 1 above)
Comment (lesteve) [inline nilearn/input_data/nifti_spheres_masker.py]: @AlexandreAbraham you haven't answered this comment.

File: nilearn/input_data/tests/test_nifti_maps_masker.py [hunk 2]
```diff
@@ -233,3 +233,33 @@ def test_nifti_maps_masker_2():
...
+
+def test_nifti_maps_masker_overlap():
+    # Test resampling in NiftiMapsMasker
+    affine = np.eye(4)
+    shape = (5, 5, 5)
+    length = 10
+
+    fmri_img, _ = generate_random_img(shape, affine=affine,
+                                      length=length)
+    no_maps = np.zeros(shape + (2,))
```
Comment (lesteve) [inline nilearn/input_data/tests/test_nifti_maps_masker.py]: For the sake of clarity, what about expanding this `no_maps` into `non_overlapping_maps` and `o_maps` into `overlapping_maps`?

Comment (lesteve) [inline nilearn/input_data/tests/test_nifti_maps_masker.py]: Same thing for _img and _masker variable, obviously.

File: nilearn/input_data/tests/test_nifti_spheres_masker.py [hunk 3]
```diff
@@ -70,3 +70,25 @@ def test_anisotropic_sphere_extraction():
...
+    # Test resampling in NiftiMapsMasker
+    affine = np.eye(4)
+    shape = (5, 5, 5)
+
+    data = np.random.random(shape + (5,))
+    fmri_img = nibabel.Nifti1Image(data, affine)
+
+    seeds = [(0, 0, 0), (2, 2, 2)]
+
+    o_masker = NiftiSpheresMasker(seeds, radius=1, allow_overlap=True)
```
Comment (lesteve) [inline nilearn/input_data/tests/test_nifti_spheres_masker.py]: Original remark about o -> overlapping, no -> non_overlapping applied to this file as well.

**Comment by lesteve** [comment]
Thanks I'll fix up the naming of the variable in master directly.

**Pull request merged**